tests/rptest/tests/transactions_test.py

```diff
@@ -0,0 +1,235 @@
+# Copyright 2022 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from rptest.services.cluster import cluster
+from ducktape.utils.util import wait_until
+from rptest.clients.types import TopicSpec
+
+import time
+
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.services.rpk_consumer import RpkConsumer
+import confluent_kafka as ck
+from rptest.clients.kafka_cat import KafkaCat
+
+import subprocess
+
+
+class TransactionsTest(RedpandaTest):
+    topics = (TopicSpec(partition_count=1, replication_factor=3),
+              TopicSpec(partition_count=1, replication_factor=3))
+
+    def __init__(self, test_context):
+        extra_rp_conf = {
+            "enable_idempotence": True,
+            "enable_transactions": True,
+            "transaction_coordinator_replication": 3,
+            "id_allocator_replication": 3,
+            "enable_leader_balancer": False,
+            "enable_auto_rebalance_on_node_add": False
+        }
+
+        super(TransactionsTest, self).__init__(test_context=test_context,
+                                               extra_rp_conf=extra_rp_conf)
+
+        self.input_t = self.topics[0]
+        self.output_t = self.topics[1]
+        self.max_records = 100
+
+    def generate_data(self, topic, num_records):
+        producer = ck.Producer({
+            'bootstrap.servers': self.redpanda.brokers(),
+        })
+
+        for i in range(num_records):
+            producer.produce(topic.name, str(i), str(i), 0)
+
+        producer.flush()
+
+    def check_consumer(self, consumer, timeout_for_consume=2):
+        def consume_records():
+            for_one_consume = 10
+            self.records = consumer.consume(for_one_consume,
+                                            timeout_for_consume)
+            return (self.records != None) and (len(self.records) != 0)
+
+        wait_until(consume_records,
```

Comment:
Let use `wait_until_result` and return the consumed records instead of setting a field. When we return data it's obvious that they are the result of the read but when we pass the result via `self.records` this connection isn't lost and a reader need to send more efforts to decipher what's going on in the test:

```
self.check_consumer(consumer1)
for record in self.records:
   ...
```

This version is more obvious:

```
records = self.consume(consumer1, max_records=10, timeout_s=2)
for record in records:
   ...
```